src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -365,13 +372,15 @@ private static unsafe void GetCursorPosition(out int left, out int top)
             {
                 // Because the CPR request/response protocol involves blocking until we get a certain
                 // response from the terminal, we want to avoid doing so if we don't know the terminal
-                // will definitely response.  As such, we start with minChars == 0, which causes the
+                // will definitely respond.  As such, we start with minChars == 0, which causes the
                 // terminal's read timer to start immediately.  Once we've received a response for
                 // a request such that we know the terminal supports the protocol, we then specify
                 // minChars == 1.  With that, the timer won't start until the first character is
                 // received.  This makes the mechanism more reliable when there are high latencies
-                // involved in reading/writing, such as when accessing a remote system.
-                Interop.Sys.InitializeConsoleBeforeRead(minChars: (byte)(s_everReceivedCursorPositionResponse ? 1 : 0), decisecondsTimeout: 10);
+                // involved in reading/writing, such as when accessing a remote system. We also extend
+                // the timeout on the very first request to 15 seconds, to account for potential latency
+                // before we know if we will receive a response.
+                Interop.Sys.InitializeConsoleBeforeRead(minChars: (byte)(s_everReceivedCursorPositionResponse ? 1 : 0), decisecondsTimeout: (byte)(s_firstCursorPositionRequest ? 150 : 10));
```

Comment:
What are the cases where a terminal won't respond to the CPR at all?  Is it acceptable to block for 15s in those cases?

Reply:
I'm not aware of any cases where that would definitely happen, but I was accounting for the possibility of a terminal that doesn't support CPR. In that case I suppose we would have a larger set of issues, as our entire scheme for checking for cursor position wouldn't work - I think I'd prefer 10 seconds or even 5 to 15, @jianyunt would a smaller value work for CloudShell?

Reply:
10 seems like a good compromise.